effekt/shared/src/main/scala/effekt/core/PolymorphismBoxing.scala

```diff
@@ -522,16 +465,14 @@ object PolymorphismBoxing extends Phase[CoreTransformed, CoreTransformed] {
         val vparams = vcoercers.map { c => ValueParam(TmpValue("coe"), transform(c.from)) }
         val bparams = bcoercers.map { c => val id = TmpBlock("coe"); BlockParam(id, transform(c.from), Set(id)) }
 
-        val inner = TmpBlock()
         val vargs = (vcoercers zip vparams).map { case (c, p) => c(Pure.ValueVar(p.id, p.tpe)) }
         val bargs = (bcoercers zip bparams).map { case (c, p) => c(Block.BlockVar(p.id, p.tpe, Set.empty)) }
         Block.BlockLit(ftparams, bparams.map(_.id), vparams, bparams,
-          Def(inner, block,
             coerce(Stmt.App(
-              Block.BlockVar(inner, block.tpe, block.capt),
+              block,
               (targs map transformArg) ++ (ftparams map core.ValueType.Var.apply),
               vargs,
-              bargs), tresult)))
+              bargs), tresult))
```

Comment:
Just to be sure [I'm sure you know.]: Can
a) Machine now handle `App(BlockLit(...), ...)`, or
b) are you ensuring this is not generated here?

I vaguely remember this being a problem (which is why the implementation is as it is).
You probably want to do b (i.e. Beta-reduce) if this runs after optimizations.

Reply:
It can handle it now, thanks for the reminder.